src/jemalloc/include/jemalloc/internal/jemalloc_internal.h.in

```diff
@@ -1041,7 +1041,7 @@ iralloct(void *ptr, size_t size, size_t extra, size_t alignment, bool zero,
 			}
 		}
 
-		return NULL;
+		return (ret != NULL) ? (ret) : (ptr);
```

Comment:
This is not the expected behavior of realloc. It SHOULD return NULL if reallocation is unsuccessful. That's why you should always use realloc in the following pattern:
```
void *ptr2 = realloc(ptr, ...);
if (ptr2 != NULL) ptr = ptr2;

```
man 3 malloc:

> The realloc() function returns ... NULL if the request fails. ... If realloc() fails, the original block is left untouched; it is not freed or moved.